plugins/kubernetes/src/components/K8sTerminal/K8sTerminalDialog.tsx

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Button,
+  createStyles,
+  Dialog,
+  DialogContent,
+  DialogTitle,
+  IconButton,
+  makeStyles,
+  Theme,
+} from '@material-ui/core';
+import CloseIcon from '@material-ui/icons/Close';
+import SubjectIcon from '@material-ui/icons/Subject';
+import React, { useState } from 'react';
+
+import { K8sTerminal, K8sTerminalProps } from './K8sTerminal';
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    dialogPaper: { minHeight: 'calc(100% - 64px)' },
+    dialogContent: { flexBasis: 0 },
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+  }),
+);
+
+/**
+ * Opens a Terminal shell to the given pod's container in a Dialog
+ *
+ * @public
+ */
+export const K8sTerminalDialog = (props: K8sTerminalProps) => {
+  const classes = useStyles();
+  const { clusterName, containerName, podName } = props;
+  const [open, setOpen] = useState(false);
+
+  const openDialog = () => {
+    setOpen(true);
+  };
+
+  const closeDialog = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <Dialog
+        maxWidth={false}
+        fullWidth
+        open={open}
+        onClose={closeDialog}
+        PaperProps={{ className: classes.dialogPaper }}
+      >
+        <DialogTitle id="dialog-title">
+          {podName} - {containerName} terminal shell on cluster {clusterName}
+          <IconButton
+            aria-label="close"
+            className={classes.closeButton}
+            onClick={closeDialog}
+          >
+            <CloseIcon />
+          </IconButton>
+        </DialogTitle>
+        <DialogContent className={classes.dialogContent}>
+          <K8sTerminal {...props} />
+        </DialogContent>
+      </Dialog>
+      <Button
+        variant="outlined"
+        aria-label="get logs"
```

Comment:
maybe a copy-paste error?
```suggestion
        aria-label="open terminal"
```